src/syft/lib/python/primitive.py

```diff
@@ -0,0 +1,122 @@
+import json
+from typing import Any
+from typing import Optional
+from typing import List
+
+from google.protobuf.reflection import GeneratedProtocolMessageType
+
+import syft as sy
+
+from ...proto.lib.python.python_primitive_pb2 import (
+    PythonPrimitive as PythonPrimitive_PB,
+)
+
+from ...core.common.serde import _deserialize
+from ...decorators.syft_decorator_impl import syft_decorator
+from ...core.store.storeable_object import StorableObject
+from ...core.common.uid import UID
+
+
+@syft_decorator(typechecking=True)
+def isprimitive(value: Any) -> bool:
+    # check primitive types first
+    if type(value) in [type(None), bool, int, float]:
+        return True
+
+    # if the type is a collection lets recursively search
+    # set cant be easily converted to json
+    if type(value) in [tuple, list]:
+        for sub in value:
+            is_subprimitive = isprimitive(value=sub)
```

Comment:
Can't we also have primitves mixed with non primitives?

Reply:
yes, there are a few Tuple(Tensor, Tensor) return types I have seen, but we need to figure out how best to mix these.

Reply:
Also for args / kwargs definitely, but this was just a quick stab at return types